src/core/observable.ts

```diff
@@ -148,8 +148,11 @@ export function reportObserved(observable: IObservable): boolean {
             }
         }
         return true
-    } else if (observable.observers.size === 0 && globalState.inBatch > 0) {
-        queueForUnobservation(observable)
+    } else if (observable.observers.size === 0) {
+        if (globalState.inBatch > 0) {
+            queueForUnobservation(observable)
+        }
+        warnAboutUntrackedObservableRead(observable.name)
```

Comment:
This warns only if there are no observers, but we want to warn anytime outside of derivation (and not in batch), right?.
The motivation behind `computedRequiresReaction` was a bit different - to avoid erasing it's cache - so it doesn't matter where the computed is accessed as far as there is at least one reaction keeping the cache alive.

Reply:
```javascript
test("warn on unsafe reads of observable 1", function() {
    try {
        mobx.configure({ observablesRequiresReaction: true })
        const x = mobx.observable({
            y: 3
        })               
 
        autorun(() => x.y); // <-- ARBITRARY OBSERVER (observers.size !== 0)

        utils.consoleWarn(() => {
            x.yy
        }, /being read outside a reactive context/)
    } finally {
        mobx.configure({ observablesRequiresReaction: false })
    }
})
```